### PR TITLE
MULE-18353: Adding test for drop event using async-try subflow

### PR DESCRIPTION
**Pull request by eze210**

### CODE REVIEW

File: integration/src/test/java/org/mule/test/integration/AsyncTryTestCase.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+@RunnerDelegateTo(Parameterized.class)
+public class AsyncTryTestCase extends AbstractIntegrationTestCase {
+
+  private String flowName;
+  private Latch latch;
+
+  @Parameterized.Parameters(name = "{0}")
+  public static Collection<Object[]> data() {
+    return Arrays.asList(new Object[][] {
+        {"Synchronous", "trigger-sync"},
```
Comment (elrodro83) [inline integration/src/test/java/org/mule/test/integration/AsyncTryTestCase.java:35]: isn't the flowname descriptive enough? why do you need the label?

File: integration/src/test/java/org/mule/test/integration/AsyncTryTestCase.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+  @Before
+  public void setupLatch() {
+    latch = new Latch();
+  }
+
+  @Test
+  public void executesTheProcessorAfterTry() throws Exception {
+    flowRunner(flowName).withVariable("latch", latch).run();
+    latch.await(20, SECONDS);
```
Comment (elrodro83) [inline integration/src/test/java/org/mule/test/integration/AsyncTryTestCase.java:63]: isnt; 20 too high? does this work with RECEIVE_TIMEOUT? (5 secs)

File: integration/src/test/resources/async-try-config.xml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            <try>
+                <raise-error type="APP:ERROR" description="An error" />
+                <error-handler>
+                    <on-error-continue enableNotifications="false" logException="false">
+                        <test:processor processingType="CPU_INTENSIVE" />
+                    </on-error-continue>
+                </error-handler>
+            </try>
+            <test:invocation-counter name="invocation-counter" />
+            <expression-component>mel:flowVars['latch'].release()</expression-component>
```
Comment (elrodro83) [inline integration/src/test/resources/async-try-config.xml:42]: avoid using deprecated stuff (mel, expression-component)

File: integration/src/test/resources/async-try-config.xml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        <try>
+            <raise-error type="APP:ERROR" description="An error" />
+            <error-handler>
+                <on-error-continue enableNotifications="false" logException="false">
+                    <test:processor processingType="CPU_INTENSIVE" />
+                </on-error-continue>
+            </error-handler>
+        </try>
+        <test:invocation-counter name="invocation-counter" />
+        <expression-component>mel:flowVars['latch'].release()</expression-component>
```
Comment (elrodro83) [inline integration/src/test/resources/async-try-config.xml:68]: avoid using deprecated stuff (mel, expression-component)

File: integration/src/test/java/org/mule/test/integration/AsyncTryTestCase.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+@Issue("MULE-18353")
+@RunnerDelegateTo(Parameterized.class)
+public class AsyncTryTestCase extends AbstractIntegrationTestCase {
+
+  private String flowName;
+  private Latch latch;
+
+  @Parameterized.Parameters(name = "{0}")
+  public static Collection<Object> data() {
```
Comment (elrodro83) [inline integration/src/test/java/org/mule/test/integration/AsyncTryTestCase.java:33]: when each parameterization is a single value like this, you can avoid the `[][]`, return a `Collection<String>` directly here

File: integration/src/test/java/org/mule/test/integration/AsyncTryTestCase.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+import java.util.Arrays;
+import java.util.Collection;
+
+import io.qameta.allure.Issue;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+
+@Issue("MULE-18353")
+@RunnerDelegateTo(Parameterized.class)
```
Comment (elrodro83) [inline integration/src/test/java/org/mule/test/integration/AsyncTryTestCase.java:26]: add the other allure annotations (feature/use case)

**Comment by eze210** [comment]
--validate

**Comment by eze210** [comment]
--validate

**Comment by eze210** [comment]
--validate

**Comment by eze210** [comment]
--validate

**Pull request merged**